### PR TITLE
[New Rule] Persistence via DirectoryService Plugin Modification

### PR DESCRIPTION
**Pull request by Samirbous**
## Issues

Resolves https://github.com/elastic/detection-rules/issues/857

## Contributor checklist

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/detection-rules/blob/main/CONTRIBUTING.md)?

### CODE REVIEW

File: rules/macos/persistence_directory_services_plugins_modification.toml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+creation_date = "2021/01/13"
+maturity = "production"
+updated_date = "2021/01/13"
+
+[rule]
+author = ["Elastic"]
+description = """
+Identifies the creation of modification of a dsplug file. The DirectoryService daemon launches on each system boot, and
+automatically reloads after crash. It scans bundles that are located in the DirectoryServices PlugIns folder and execute
+them. Adversaries may abuse this feature for persistence.
```
Comment (bm11100) [inline rules/macos/persistence_directory_services_plugins_modification.toml:11]:
```suggestion
Identifies the creation or modification of a DirectoryService PlugIns (dsplug) file. The DirectoryService daemon launches on each system boot and
automatically reloads after crash. It scans and executes bundles that are located in the DirectoryServices PlugIns folder and can be abused by adversaries to maintain persistence.
```

Comment (bm11100) [reply rules/macos/persistence_directory_services_plugins_modification.toml:11]: re-lint

**Review by bm11100 — APPROVED** [review]
🍕

File: rules/macos/persistence_directory_services_plugins_modification.toml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+"""
+false_positives = ["Trusted DirectoryService Plugins"]
```
Comment (bm11100) [inline rules/macos/persistence_directory_services_plugins_modification.toml:13]:
```suggestion
false_positives = ["Trusted DirectoryService PlugIns"]
```

File: rules/macos/persistence_directory_services_plugins_modification.toml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+creation_date = "2021/01/13"
+maturity = "production"
+updated_date = "2021/01/13"
+
+[rule]
+author = ["Elastic"]
+description = """
+Identifies the creation or modification of a DirectoryService PlugIns (dsplug) file. The DirectoryService daemon launches on each system boot and
+automatically reloads after crash. It scans and executes bundles that are located in the DirectoryServices PlugIns folder and can be abused by adversaries to maintain persistence.
+"""
```
Comment (threat-punter) [inline rules/macos/persistence_directory_services_plugins_modification.toml:11]:
```suggestion
description = """
Identifies the creation or modification of a DirectoryService PlugIns (dsplug) file. The DirectoryService 
daemonlaunches on each system boot and automatically reloads after crash. It scans and executes bundles that are
located in the DirectoryServices PlugIns folder and can be abused by adversaries to maintain persistence.
"""
```

**Review by threat-punter — APPROVED** [review]

**Pull request merged**